ci_framework/plugins/action/ci_make.py

```diff
@@ -0,0 +1,57 @@
+# This specific action plugin needs a version of community.general collection
+# providing this patch:
+# https://github.com/ansible-collections/community.general/pull/6160
+# While the "make" command will actually run, it won't be able to generate
+# the needed file.
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+
+from ansible.plugins.action import ActionBase
+from ansible.errors import AnsibleActionFail, AnsibleError
+from ansible.utils.display import Display
+
+ERR_UPDATE_COLLECTION = '''"command" not found in community.genera.make output.
+Please update collection. The expected reproducer file will NOT be generated!
+'''
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        super(ActionModule, self).run(tmp, task_vars)
+        module_args = self._task.args.copy()
+
+        if 'output_dir' not in module_args:
+            raise AnsibleActionFail('output_dir parameter is missing')
+
+        output_dir = module_args.pop('output_dir')
+        # Check for directory
+        if not os.path.exists(output_dir):
+            raise AnsibleError("%s doesn't exist" % output_dir)
+        if not os.path.isdir(output_dir):
+            raise AnsibleError("%s isn't a directory" % output_dir)
+        if not os.access(output_dir, os.W_OK):
+            raise AnsibleError("%s isn't writable" % output_dir)
```

Comment:
This can be be combined in one check. It could make sense to move this logic in an exception block. There it would be triggered only if writing to dir fails, and provide information to the user about the reason why.